### PR TITLE
Teams initial support

### LINKED ISSUE #144: Granting teams access to workspaces
Not really a bug, but rather a feature request.
Want to grant teams access to workspaces. And would like a `data source` for getting the teams.
Cant see it existing already.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform Version

`0.1.1`

### Affected Resource(s)

* resource missing

### Terraform Configuration Files

Want to do something like:

```tf
data "prefect_teams" "all_teams" {
 
}

resource "prefect_workspace_access_team" "viewers" {
  role = "viewer"
  team = # data.prefect_teams.all_teams filtered on some condition
}
resource "prefect_workspace_access_team" "viewers" {
  role = "owner"
  team = # data.prefect_teams.all_teams filtered on some condition
}

# alt. extending the existing prefect_workspace_access-resource

```

### PR DESCRIPTION
**Pull request by triedandtested-dev**
Resolves #144 

- [x] workspace_access - support for team.
- [x]  teams data source
- [x]  team data source

### CODE REVIEW

**Comment by triedandtested-dev** [comment]
@gabcoyne I removed the unsupported checker from the linter should be good now

**Comment by mch-sb** [comment]
Nice!👌 This would help out us quite a bit also!

File: .golangci.yaml [hunk 1]
```diff
@@ -155,7 +155,6 @@ linters-settings:
       - sliceClear
       - sloppyLen
       - sloppyReassign
-      - sloppyTestFuncName
```
Comment (parkedwards) [inline .golangci.yaml:158]: 👍

File: internal/api/teams.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+type Team struct {
+	BaseModel
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+// TeamFilter defines the search filter payload
+// when searching for team by name.
+// example request payload:
+// {"teams": {"name": {"any_": ["test"]}}}.
```
Comment (parkedwards) [inline internal/api/teams.go:22]: nice

File: internal/provider/resources/workspace_access.go [hunk 3]
```diff
@@ -87,9 +87,9 @@ func (r *WorkspaceAccessResource) Schema(_ context.Context, _ resource.SchemaReq
 			},
 			"accessor_type": schema.StringAttribute{
 				Required:    true,
-				Description: "USER or SERVICE_ACCOUNT",
+				Description: "USER | SERVICE_ACCOUNT | TEAM",
```
Comment (parkedwards) [inline internal/provider/resources/workspace_access.go:90]: 👍

**Review by parkedwards — APPROVED** [review]
amazing work! I left a non-blocking suggestion for the docs section, but this looks great. thank you for doing this!

**Comment by parkedwards** [comment]
ok this has to do with how we set up our workflow triggers - GHA won't insert Secrets from a forked workflow for obvious reasons.

I'm going to merge this in, but also sanitize our main branch in case there are failures.  thanks for the contribution @triedandtested-dev

**Pull request merged**